dev/tools/dartdoc.dart

```diff
@@ -211,16 +216,62 @@ void createFooter(String footerPath) {
     gitBranchOut].join(' '));
 }
 
+/// Recursively copies `srcDir` to `destDir`, invoking [onFileCopied] if
+/// specified for each source/destination file pair.
+///
+/// Creates `destDir` if needed.
+void copyDirectorySync(Directory srcDir, Directory destDir, [void onFileCopied(File srcFile, File destFile)]) {
+  if (!srcDir.existsSync())
+    throw Exception('Source directory "${srcDir.path}" does not exist, nothing to copy');
+
+  if (!destDir.existsSync())
+    destDir.createSync(recursive: true);
+
+  for (FileSystemEntity entity in srcDir.listSync()) {
+    final String newPath = path.join(destDir.path, path.basename(entity.path));
+    if (entity is File) {
+      final File newFile = File(newPath);
+      entity.copySync(newPath);
+      onFileCopied?.call(entity, newFile);
+    } else if (entity is Directory) {
+      copyDirectorySync(entity, Directory(newPath));
+    } else {
+      throw Exception('${entity.path} is neither File nor Directory');
+    }
+  }
+}
+
+void copyAssets() {
+  final Directory assetsDir = Directory(path.join(kPublishRoot, 'assets'));
+  if (assetsDir.existsSync()) {
+    assetsDir.deleteSync(recursive: true);
+  }
+  copyDirectorySync(
+    Directory(path.join(kDocsRoot, 'assets')), Directory(path.join(kPublishRoot, 'assets')),
+      (File src, File dest) {
+      print('Copying ${src.path} to ${dest.path}');
```

Comment:
Technically, it's "Copied".  Or you should change the parameter name and semantics.